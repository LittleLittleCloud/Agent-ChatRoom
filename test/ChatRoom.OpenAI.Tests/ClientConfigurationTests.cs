using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Approvers;
using ApprovalTests.Core;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ChatRoom.OpenAI;
using ChatRoom.SDK;
using FluentAssertions;
using Json.Schema;
using Json.Schema.Generation;
using Moq;
using Xunit;

namespace ChatRoom.OpenAI.Tests;

public class ClientConfigurationTests
{
    private readonly JsonSerializerOptions _jsonSerializerOptions = new JsonSerializerOptions { WriteIndented = true };
    
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public void VerifyConfigurationSchema()
    {
        var schema = new JsonSchemaBuilder()
            .FromType<Configuration>()
            .Build();

        var schemaFileName = "chatroom_openai_configuration_schema.json";
        var schemaFilePath = Path.Join("template", "chatroom.openai", schemaFileName);
        var schemaFile = File.ReadAllText(schemaFilePath);

        var json = JsonSerializer.Serialize(schema, new JsonSerializerOptions { WriteIndented = true });
        Approvals.Verify(json);
        schemaFile.Should().BeEquivalentTo(json);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public void ConfigurationLoadTest()
    {
        var config = new OpenAIAgentConfiguration
        {
            LLMConfiguration = new OpenAIClientConfiguration
            {
                LLMType = LLMType.AOAI,
                OpenAIApiKey = "test",
                OpenAIModelId = "test",
                AzureOpenAIEndpoint = "test",
                AzureOpenAIKey = "test",
                AzureOpenAIDeployName = "test",
                ThirdPartyLLMEndpoint = "test",
                ThirdPartyLLMKey = "test"
            },
            SystemMessage = "test",
            Description = "test",
            Name = "test"
        };

        var json = JsonSerializer.Serialize(config, _jsonSerializerOptions);
        Approvals.Verify(json);
        config = JsonSerializer.Deserialize<OpenAIAgentConfiguration>(json)!;

        json.Should().BeEquivalentTo(JsonSerializer.Serialize(config, _jsonSerializerOptions));
    }
}
