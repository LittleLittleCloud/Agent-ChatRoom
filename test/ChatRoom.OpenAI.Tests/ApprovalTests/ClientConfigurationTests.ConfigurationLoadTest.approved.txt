{
  "llm_config": {
    "llm_type": "AOAI",
    "openai_api_key": "test",
    "openai_model_id": "test",
    "azure_openai_endpoint": "test",
    "azure_openai_key": "test",
    "azure_openai_deploy_name": "test",
    "third_party_llm_endpoint": "test",
    "third_party_llm_key": "test",
    "third_party_llm_model_id": null
  },
  "system_message": "test",
  "agent_description": "test",
  "name": "test"
}